app/src/main-process/squirrel-updater.ts

```diff
@@ -72,33 +79,65 @@ function getBinPath(): string {
  * rewrite the trampoline to point to the new, version-specific path. Bingo
  * bango Bob's your uncle.
  */
-async function writeCLITrampoline(): Promise<string> {
-  const binPath = getBinPath()
-  const appFolder = Path.resolve(process.execPath, '..')
-  const versionedPath = Path.relative(
+async function writeBatchScriptCLITrampoline(binPath: string): Promise<void> {
+  const versionedPath = resolveVersionedPath(
     binPath,
-    Path.join(appFolder, 'resources/app/static/github.bat')
+    'resources/app/static/github.bat'
   )
-  const trampline = `@echo off\n"%~dp0\\${versionedPath}" %*`
+  const trampoline = `@echo off\n"%~dp0\\${versionedPath}" %*`
   const trampolinePath = Path.join(binPath, 'github.bat')
-  return new Promise<string>((resolve, reject) => {
+  return new Promise<void>((resolve, reject) => {
     Fs.ensureDir(binPath, err => {
       if (err) {
         reject(err)
         return
       }
 
-      Fs.writeFile(trampolinePath, trampline, err => {
+      Fs.writeFile(trampolinePath, trampoline, err => {
         if (err) {
           reject(err)
         } else {
-          resolve(binPath)
+          resolve()
         }
       })
     })
   })
 }
 
+async function writeShellScriptCLITrampoline(binPath: string): Promise<void> {
+  const versionedPath = resolveVersionedPath(
+    binPath,
+    'resources/app/static/github.sh'
+  )
+  const trampoline = `#!/usr/bin/env bash
+  DIR="$( cd "$( dirname "\$\{BASH_SOURCE[0]\}" )" && pwd )"
+  sh "$DIR/${versionedPath}" "$@"`
+  const trampolinePath = Path.join(binPath, 'github')
+  return new Promise<void>((resolve, reject) => {
+    Fs.ensureDir(binPath, err => {
```

Comment:
Should we use `mkdirIfNeeded` from `file-system` instead here?

Reply:
Heck yeah we should! :metal: